python/cutlass/__init__.py

```diff
@@ -156,10 +156,33 @@ def get_memory_pool():
     return this.memory_pool
 
 
-from cuda import cuda, cudart
+cuda = lazy_import("cuda.cuda")
+cudart = lazy_import("cuda.cudart")
 
 this._device_id = None
+
+def check_cuda_versions():
+    # Strip any additional information from the CUDA version
+    _cuda_version = cuda.__version__.split("rc")[0]
+    # Check that Python CUDA version exceeds NVCC version
+    _nvcc_version = cutlass.nvcc_version()
+    _cuda_list = _cuda_version.split('.')
+    _nvcc_list = _nvcc_version.split('.')
+    for val_cuda, val_nvcc in zip(_cuda_list, _nvcc_list):
+        if int(val_cuda) < int(val_nvcc):
+            raise Exception(f"Python CUDA version of {_cuda_version} must be greater than or equal to NVCC version of {_nvcc_version}")
```

Comment:
Do not raise BaseException class. Raise ImportError, RuntimeError, SystemError, AssertionError or something more specific.